packages/driver/src/cypress/mocha.ts

```diff
@@ -37,6 +37,89 @@ delete (window as any).Mocha
 
 export const SKIPPED_DUE_TO_BROWSER_MESSAGE = ' (skipped due to browser)'
 
+// NOTE: 'calculateTestStatus' is marked as an individual function to make functionality easier to test.
+export const calculateTestStatus = function (strategy?: 'detect-flake-and-pass-on-threshold' | 'detect-flake-but-always-fail', options?: any) {
```

Comment:
```suggestion
export function calculateTestStatus (test: Mocha.Test, strategy?: 'detect-flake-and-pass-on-threshold' | 'detect-flake-but-always-fail', options?: any) {
```

I think this signature would read better than using the `this` keyword, especially since this is detached from the Test class.

Then, of course, would need to change `this` to `test` within the rest of the function definition.

Reply:
This change makes sense to me and added in 3ec51b3ce8. I added a few `ts-expect-error` for our custom properties on the test runnable, but overall I think this reads a lot better!